examples/ssd/ssdmobilenet.cpp

```diff
@@ -0,0 +1,111 @@
+// Tencent is pleased to support the open source community by making ncnn available.
+//
+// Copyright (C) 2017 THL A29 Limited, a Tencent company. All rights reserved.
+//
+// Licensed under the BSD 3-Clause License (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// https://opensource.org/licenses/BSD-3-Clause
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#include <stdio.h>
+#include <algorithm>
+#include <vector>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <iostream>
+
+#include "net.h"
+
+struct Object{
+    cv::Rect rec;
+    int label_id;
+    float prob;
+};
+
+static int detect_mobilenet(cv::Mat& raw_img, float show_threshold)
+{
+    ncnn::Net mobilenet;
+    /*
+     * model is  converted from https://github.com/chuanqi305/MobileNet-SSD
+     * and can be downloaded from https://drive.google.com/open?id=0ByaKLD9QaPtucWk0Y0dha1VVY0U
+     */
+    int img_h = raw_img.size().height;
+    int img_w = raw_img.size().width;
+    mobilenet.load_param("mobilenet_ssd_voc_ncnn.param");
+    mobilenet.load_model("mobilenet_ssd_voc_ncnn.bin");
+    auto input_size = 300;
+    ncnn::Mat in = ncnn::Mat::from_pixels_resize(raw_img.data, ncnn::Mat::PIXEL_BGR, raw_img.cols, raw_img.rows, input_size, input_size);
+
+    const float mean_vals[3] = {127.5f, 127.5f, 127.5f};
+    const float norm_vals[3] = {1.0/127.5,1.0/127.5,1.0/127.5};
+    in.substract_mean_normalize(mean_vals, norm_vals);
+
+    ncnn::Mat out;
+
+    for(int i=0;i<1;i++)
+    {
+        ncnn::Extractor ex = mobilenet.create_extractor();
+        ex.set_light_mode(true);
+        //ex.set_num_threads(4);
+        ex.input("data", in);
+        ex.extract("detection_out",out);
+
+    }
+
+    printf("%d %d %d\n", out.w, out.h, out.c);
+    std::vector<Object> objects;
+    for (int ih=0;ih<out.h;ih++)
+    {
+        Object object;
+        for(int iw=0;iw<out.w;iw++)
+        {
+            auto value = out.data[iw+ih*out.w];
```

Comment:
out.w is always six, so it's fine to write const float* values = out.row(ih); and use values[0] values[1] .... instead of the following six conditions
do not use auto keyword